### PR TITLE
Report additions (JJ)

### PR DESCRIPTION
**Pull request by jjkester**
Not final, will add more.

Also, will rebase before merge.

### CODE REVIEW

**Comment by wandernauta** [comment]
:+1: Would you like to merge this or are you looking for feedback?

**Comment by jjkester** [comment]
Right now feedback, plus a place to drop in additions/changes/stuff.

File: Report/Chapters/Design.tex [hunk 1]
```diff
@@ -8,5 +8,50 @@ \section{Languages and libraries}
 
 \section{High-level architecture}
 
-% Frontend/backend split
-% Communication between frontend and backend (when we've got that settled)
+Viskell consists of two parts, a front end and a back end.
+The front end is responsible for providing a user interface, handling user input, displaying error messages, and communicating with the back end.
+The back end consists of a representation for Haskell code in Java, an interface to GHCi and functionality to make working with the back end easier.
+This separation is made to make development and debugging for the part that generates Haskell code easier.
+It also makes using the user interface for a different purpose a bit easier, although the user interface is not specifically designed for this purpose.
```
Comment (i3anaan) [inline Report/Chapters/Design.tex]: although the user interface is not specifically designed for *other* purpose.

File: Report/Chapters/Design.tex [hunk 2]
```diff
@@ -8,5 +8,50 @@ \section{Languages and libraries}
...
+A result of our design choices is that our Java program needs to have a basic understanding of the Haskell programming
+language. For this purpose we implemented two tree structures - one for expressions, one for types - supported by an
+interface for GHCi, a type signature parser and a catalog.
+
+\subsection{Communication with GHCi}
+\index{GHCi}
+
+The base of the back-end is an interface for communication with GHCi. This interface allows us to run Haskell code
+generated by the users of our application. As implementing a complete type checker for Haskell is a major challenge this
+interface can also be used to check a program for faults.
```
Comment (i3anaan) [inline Report/Chapters/Design.tex]: As implementing a complete type checker for Haskell is a major challenge this interface can also be used to check a program for faults.

I don't understand what you are trying to say here.

File: Report/Chapters/Design.tex [hunk 3]
```diff
@@ -8,5 +8,50 @@ \section{Languages and libraries}
...
+
+\subsection{Type system and type checker}
+
+The error messages from GHCi are hard to parse and do not always provide detailed information about where the error
+occured. Therefore we implemented our own type checker. This type checker is designed to catch most of the type errors
+which are in the design scope of the project and not raise false negatives. Pushing Haskell code to GHCi is still needed
+to be sure that the code compiles (and will always be needed to catch runtime errors).
+
+Internally Haskell types are represented as instances of a class. These instances can be tied together to form a
```
Comment (i3anaan) [inline Report/Chapters/Design.tex]: Internally in our java code....

or something like that

File: Report/Chapters/Design.tex [hunk 4]
```diff
@@ -8,5 +8,50 @@ \section{Languages and libraries}
...
+tree-like structure. This approach is chosen because it is easy to work with class instances and nested types form
+a tree-like structure. \index{type tree}
+
+The type checker is implemented using Hindley-Milner type inference. We use this algorithm to (try to) unify two
+types. If the types cannot be unified, there exists a type error. The algorithm is described in detail by
+\cite{borisov}. \index{Hindley-Milner} \index{type inference}
+
+\subsection{Expressions}
+
+Haskell expressions are represented in a tree. By design each function can have only one argument. Functions with more
```
Comment (i3anaan) [inline Report/Chapters/Design.tex]: We represented Haskell expressions in a tree ...

File: Report/Chapters/Design.tex [hunk 5]
```diff
@@ -8,5 +8,50 @@ \section{Languages and libraries}
...
+arguments are represented as a function with a single argument that produces another function with the other arguments.
```
Comment (i3anaan) [inline Report/Chapters/Design.tex]: that produces another function with another argument and so on.

(Currently it is as if the second function does accept more than 1 argument)

File: Report/Chapters/Design.tex [hunk 6]
```diff
@@ -8,5 +8,50 @@ \section{Languages and libraries}
 
 \section{High-level architecture}
```
Comment (i3anaan) [inline Report/Chapters/Design.tex]: Im missing the argumentation as to why we decided to implement our own type system as opposed to using the error messages from GHCi

File: Report/Chapters/Introduction.tex [hunk 7]
```diff
@@ -2,7 +2,6 @@ \chapter{Introduction}
 
 Programming has always been an activity which is done on a computer with a keyboard. Despite the existence of different visual programming languages, like Blender, there are no programs in which well known programming languages like Java, Haskell or C(++) can be visually implemented.
```
Comment (i3anaan) [inline Report/Chapters/Introduction.tex]: Is Blender a visual programming language?
Do you perhaps mean the visual programming language used within Blender?

File: Report/Chapters/Introduction.tex [hunk 8]
```diff
@@ -2,7 +2,6 @@ \chapter{Introduction}
...
 
-In this project we have (attempted) to make an application in which the programming language Haskell can be visually implemented.
-%aanvullen als het project af is
+In this project we have (attempted) to make an application in which the programming language Haskell can be visually implemented. We named our approach ``Viskell''. Viskell tries to aide developers with real-time feedback on Haskell programs. This allows developers to spot type errors directly, without needing to have a compilable program, and provides a visual insight into the program. Also, the intermediate output of a function can be inspected at runtime. Viskell also allows developers to collaborate and work together on a large multi-touch screen. Other common touch interfaces as found on devices like the Microsoft Surface (Pro), are supported as well. Because of the user interface with the touch capabilities developers can easily manipulate the code without having a keyboard or mouse around, so programming on the go is a better experience. Besides that programming with Viskell should be more fun and more ergonomical because seating positons can be changed.
```
Comment (i3anaan) [inline Report/Chapters/Introduction.tex]: > so programming on the go is a better experience.

Not so sure about that tbh

The entire section has lots of claims that might not be fully justified

File: Report/Chapters/Process.tex [hunk 9]
```diff
@@ -20,7 +20,7 @@ \subsection{Process tasks}
 
 \subsection{Implementation and design tasks}
 
-For the very first part of the project, Martijn, Kristel and Derk were assigned to start working on the program's front end design. Wander and Jan-Jelle were to tackle the beginnings of a usable back end, that is, getting the program designed in the front end to run. This split kept existing during most of the project.
+For the very first part of the project, Martijn, Kristel and Derk were assigned to start working on the program's front end design. Wander and Jan-Jelle were to tackle the beginnings of a usable back end, that is, getting the program designed in the front end to run. This split kept existing during most of the project because of the familiarity with the code.
```
Comment (i3anaan) [inline Report/Chapters/Process.tex]: > This split kept existing during most of the project because of the familiarity with the code.

Not just code, familiarity with the subjects we already worked on, can't really think of the correct word, but its more than just code.

File: Report/Chapters/Process.tex [hunk 10]
```diff
@@ -28,17 +28,17 @@ \section{Code tools}
 
 \subsection{Maven}
 
-Larger Java projects usually have dependencies. These dependencies can also have dependencies. To make sure every team member was using the same versions of the libraries the project was no doubt going to use, we decided to use Maven, a Java build tool. Using Maven as a standardized build process also allowed team members to work from their preferred environment: the back end crew preferred IntelliJ IDEA by Jetbrains, while the front end developers worked mostly in Eclipse. \index{Maven}
+Larger Java projects usually have dependencies. These dependencies can also have dependencies. To make sure every team member was using the same versions of the libraries the project was no doubt going to use, we decided to use Maven, a Java build tool. Using Maven as a standardized build process also allowed team members to work from their preferred environment: the back end crew preferred IntelliJ IDEA by JetBrains, while the front end developers worked mostly in Eclipse. \index{Maven}
```
Comment (i3anaan) [inline Report/Chapters/Process.tex]: These dependencies can *in turn* also have dependencies.

> To make sure every team member was using the same versions of the libraries the project was no doubt going to use, we decided to use Maven, a Java build tool

Hard to read sentence

File: Report/Chapters/Process.tex [hunk 11]
```diff
@@ -28,17 +28,17 @@ \section{Code tools}
...
 
 \subsection{Git, GitHub and GitHub Flow}
 
 For the version control system, we decided on git, with the central repository stored on GitHub. \index{git} \index{GitHub}
 
-As for a branching model, we decided on GitHub Flow\cite{githubflow}. In this model, small, independent changes are published to their own branches, tested, checked, then proposed as a Pull Request. At this point the person responsible for code integration (either Wander or Jan-Jelle) reviews the code and merge it into the master branch. \index{GitHub Flow}
+As for a branching model, we decided on GitHub Flow\cite{githubflow}. In this model, small, independent changes are published to their own branches, tested, checked, then proposed as a Pull Request. At this point the person responsible for code integration (either Wander or Jan-Jelle) reviews the code and merge it into the master branch. \index{GitHub Flow} Additionally, GitHub is used as issue tracker.
```
Comment (i3anaan) [inline Report/Chapters/Process.tex]: Additionally, GitHub *was* used as issue tracker. ?
Not 100% sure  what time you are using, but I have a feeling this is inconsitent with your previous usage of time.

File: Report/Chapters/Process.tex [hunk 12]
```diff
@@ -28,17 +28,17 @@ \section{Code tools}
...
 
 \subsection{Travis}
 
-It is no use having automated unit tests if they are never executed. To make sure we would get alerted to broken tests quickly, we decided to use Travis CI with the project to build every change and run the unit tests. Travis automatically sends email notifications when a build fails. It also integrates with GitHub's pull request feature, showing the build status there as well to help keep broken code out of the master branch. \index{Travis} \index{continuous integration}
+It is no use having automated unit tests if they are never executed. To make sure we would get alerted to broken tests quickly, we decided to use Travis CI with the project to build every change and run the unit tests. Travis automatically sends email notifications when a build fails. It also integrates with GitHub's pull request feature, showing the build status there as well to help keep broken code out of the master branch. Travis has proven to be a very useful tool. \index{Travis} \index{continuous integration}
```
Comment (i3anaan) [inline Report/Chapters/Process.tex]: > to build every change and run the unit tests

Every branch

File: Report/Chapters/Requirements.tex [hunk 13]
```diff
@@ -1,26 +1,76 @@
 \chapter{Requirements Analysis}
 
-The first step in creating an application or program is defining the requirements. In our discussions with the client, he emphasized that the main goal of this project was making a user-friendly interface.
+The first step in creating an application or program was defining the requirements. In order to do this, we planned meetings with our client. Against our expectations the client did not have many demands to the project. He wanted us to be free in the design and implementation. The main goal of the project was not creating a program, which is 100\% finished and could be used right away, but to explore the possibilities of the client's ideas (a proof of concept).
```
Comment (i3anaan) [inline Report/Chapters/Requirements.tex]: > Against our expectations the client did not have many demands to the project. He wanted us to be free in the design and implementation.

Misschien kom je hier later op terug, maar dit bleek dus ook weer niet zo te zijn (hij wees bijvoorbeeld ons eerste idee voor een UI af)

File: Report/Chapters/Requirements.tex [hunk 14]
```diff
@@ -1,26 +1,76 @@
...
-In this project the focus lies mainly on users who already know how to programme in Haskell. %nog aanvulling nodig,..
+\section{Target audience}
 
-\subsection{Requirements}
-For this project, the requirements can be devided in three subgroups, namely: visual requirements, functional requirements and multi-touch requirements. In the following subsections each of the groups will be discussed. In some cases a requirement fits in multiple groups; in these cases the best fitting group has been chosen to contain this requirement.
+How to divide the requirements in necessary and desirable requirements depends on the kind of target audience for which the application is designed. \index{audience} \index{target audience}
 
+In this project the focus lies mainly on users who already know how to program in Haskell. This includes people who are just starting with Haskell.
+The program should aid these users in creating an overview of their program and informing them about (possible) mistakes without the user having to write a complete program and then compile it.
```
Comment (i3anaan) [inline Report/Chapters/Requirements.tex]: remove '(possible)'

File: Report/Chapters/Requirements.tex [hunk 15]
```diff
@@ -1,26 +1,76 @@
...
 
-As discussed earlier, the user experience is a very important aspect in this project. With this in mind, the visual aspects of the application are important. De visual requirements are the following:
+\section{Hardware}
+
+The program will be designed to use on a large multi-touch screen with a screen resolution of 1920x1080 pixels, but should also work with a smaller (for example laptop) screen and a mouse. The program should be efficient enough to run on average modern-day hardware. The client provided a desktop computer and multi-touch screen for reference.
+
+\section{Requirements}
+
+For this project, the requirements can be divided in three subgroups, namely: visual requirements, functional requirements and multi-touch requirements. In the following subsections each of the groups will be discussed. In some cases a requirement fits in multiple groups; in these cases the best fitting group has been chosen to contain this requirement.
```
Comment (i3anaan) [inline Report/Chapters/Requirements.tex]: I'm not so convinced the split in 3 parts is a correct split, we should at the very least provide some argumentation as to why this split was made.

File: Report/Chapters/Testplan.tex [hunk 16]
```diff
@@ -1,7 +1,21 @@
 \chapter{Test Plan}
 
 To test the finished product, we used (automatic) unit tests as well as (manual) acceptance tests.
+The acceptance tests are focsued on the front end as it is difficult to use automated unit tests for this purpose.
```
Comment (i3anaan) [inline Report/Chapters/Testplan.tex]: > focsued

Typo

Aside from that, you should turn the sentence around:
Since unit tests are hard for front end we used acceptance tests.

File: Report/Chapters/Design.tex (hunk 1 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Reworded to "... specifically designed to be reusable."

File: Report/Chapters/Design.tex (hunk 6 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: See "type system and type checker" section.

File: Report/Chapters/Introduction.tex (hunk 7 above)
Comment (jjkester) [inline Report/Chapters/Introduction.tex]: You could see Blender as a programming language that is purely visual.

I think that the claim is right, but I am not 100% sure. But I like the 'in your face'-ness of it. I think I want to keep it this way.

Comment (wandernauta) [inline Report/Chapters/Introduction.tex]: Respectfully disagree: Blender is a 3D modeling tool. *One* way of using a *part* of that tool (the nodes interface to the compositor) has some superficial (visual) similarities to what we're doing, but I wouldn't call it a programming language, and doing so could perhaps confuse our readers.

A better example might be Pure Data (Pd), which is both niche enough for people never to have heard of, close enough to what we're doing, and far enough from something like Java for the sentence to make sense.

File: Report/Chapters/Introduction.tex (hunk 8 above)
Comment (jjkester) [inline Report/Chapters/Introduction.tex]: On the 'programming on the go' part: replaced 'is' with 'becomes', which should clarify my point.

File: Report/Chapters/Process.tex (hunk 9 above)
Comment (jjkester) [inline Report/Chapters/Process.tex]: I know, but I realy don't know how to reword this to reflect that. I think this is an understandable explanation.

File: Report/Chapters/Requirements.tex (hunk 13 above)
Comment (jjkester) [inline Report/Chapters/Requirements.tex]: This is a point we can discuss tomorrow.

File: Report/Chapters/Requirements.tex (hunk 14 above)
Comment (jjkester) [inline Report/Chapters/Requirements.tex]: I do not agree. The type checker might indicate something as a mistake, but it might be intended. Also, we do not display an error on an input which is not connected, but doing this was an option. So not everything we display as an error is an actual error since the developer might have intended it (because he plans to change the program elsewhere). Same holds for IDEs like Eclipse and IntelliJ (the curly lines).

File: Report/Chapters/Requirements.tex (hunk 15 above)
Comment (jjkester) [inline Report/Chapters/Requirements.tex]: Why? This is just a split to put some things into context, not because things only apply to the subgroup they're in.

File: Report/Chapters/Design.tex [hunk 17]
```diff
@@ -1,17 +1,103 @@
...
+\subsubsection{Blocks and lines}
+
+We chose to represent Haskell functions by function blocks since this is a logical element for users, a combination of blocks forms a program.
+Each block consists of a function name and its associated arguments. With each argument comes an anchor node to which other block can be connected through lines.
+
+In Haskell, values are being passed to functions without being saved to memory.
+The lines between blocks represent this flow.
+Thus, a program can be built like an electronic circuit.
+
+We chose to make a distinction between three kinds of blocks.
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: We've got a few more than three now.

File: Report/Chapters/Design.tex [hunk 18]
```diff
@@ -1,17 +1,103 @@
...
+The first kind in an input block, which represents a value.
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: It's called a Value Block in the menu.

You could underline the fact that it is a constant value here, i.e. you would use one where you would use a constant literal in plain ol' Haskell.

File: Report/Chapters/Design.tex [hunk 19]
```diff
@@ -1,17 +1,103 @@
...
+This input block can be connected to other blocks to use the value as an input.
+Besides an input block that takes a string as input and automatically detects the type, there is also a slider available.
+
+The second kind of block is a function block. This block has a number of inputs and a single output, and defines the number of inputs that are going to be used.
+The number of inputs can be changed by moving a knot over the function arguments.
+This allows for passing functions as input for other functions.
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: I don't know if this is partial application or currying, but you could mention one of those here.

File: Report/Chapters/Design.tex [hunk 20]
```diff
@@ -1,17 +1,103 @@
...
+An error is represented as a colored line and colored type in the target block.
+An icon is displayed as well to help color blind users.
+
+\subsubsection{Menus}
+
+During the design process many ideas for the menus came up.
+One of the first ideas was to have a circular menu. \label{circular_menu}
+The advantages of the circle menu would be that the orientation of the device would not have any influence on the usability of the menu.
+This makes using the program on a large multi-touch table easier.
+However, the implementation of the menu would take too much time and the idea was therefore dismissed.
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: Do we have a picture of the circle menu? A sketch?

File: Report/Chapters/Design.tex [hunk 21]
```diff
@@ -1,17 +1,103 @@
...
+The final design has two separate menus, one for adding new blocks and one for options on existing blocks.
+
+The primary menu uses `drawers' for the categories of the functions.
+This way it is easy to find functions even without an extensive knowledge of Haskell.
+In- and ontput blocks and the function definition block can be created using buttons at the bottom of the menu.
+This location is chosen to make them quickly available.
+The primary menu can be opened multiple times at once and moved around the screen so multiple users can work on a single program.
+
+The context menu is compact so is does not take up too much space.
+It contains clear icons which represent different actions.
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: Could use examples (in this case, one example: cut).

File: Report/Chapters/Design.tex [hunk 22]
```diff
@@ -1,17 +1,103 @@
...
+Java 8 is chosen because it, unlike Java 7, still receives public updates at the moment of writing.
+Language level 8 is chosen to make use of some of the new features in this version.
+
+Viskell uses TactileFX, a JavaFX touch screen library developed at the University of Twente.
+The client wanted the program to use this library, which also means that Viskell had to use Java and JavaFX.
+
+Furthermore, the following libraries have been used:
+
+\begin{itemize}
+	\item Antlr 4 (http://www.antlr.org/)
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: There's a neat `\url` construction that turns these into hyperlinks in the resulting PDF.

File: Report/Chapters/Design.tex [hunk 23]
```diff
@@ -1,17 +1,103 @@
...
+	\item JFXtras (http://jfxtras.org/)
+\end{itemize}
 
 \section{High-level architecture}
 
-% Frontend/backend split
-% Communication between frontend and backend (when we've got that settled)
+Viskell consists of two parts, a front end and a back end.
+The front end is responsible for providing a user interface, handling user input, displaying error messages, and communicating with the back end.
+The back end consists of a representation for Haskell code in Java, an interface to GHCi and functionality to make working with the back end easier.
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: If this is the first mention of GHCi, add a reference.

File: Report/Chapters/Design.tex [hunk 24]
```diff
@@ -1,17 +1,103 @@
...
+This separation is made to make development and debugging for the part that generates Haskell code easier.
+It also makes using the user interface for a different purpose a bit easier, although the user interface is not specifically designed to be reusable.
```
Comment (wandernauta) [inline Report/Chapters/Design.tex]: Or, to put it more positively, the user interface has been specifically designed for our use case.

File: Report/Chapters/Design.tex (hunk 18 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Reworded.

File: Report/Chapters/Design.tex (hunk 19 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Fixed.

File: Report/Chapters/Implementation.tex [hunk 25]
```diff
@@ -1,11 +1,8 @@
 \chapter{Implementation}
+\label{chap:implementation}
```
Comment (wandernauta) [inline Report/Chapters/Implementation.tex]: Good catch.

File: Report/Chapters/Design.tex (hunk 20 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Yes, will include it later on. I was going to add all images in one go.

File: Report/Chapters/Design.tex (hunk 21 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Added delete example, since that is working.

File: Report/Chapters/Design.tex (hunk 22 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Fixed.

File: Report/Chapters/Design.tex (hunk 23 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Added

File: Report/Chapters/Process.tex [hunk 26]
```diff
@@ -46,6 +46,8 @@ \section{Process tools}
 
 \subsection{Trello}
 
+The layout of a Trello project with horizontally placed containers which hold vertically aligned cards makes it a good tool for maintaining sprints. Trello also supports check-off lists, assigning people to cards and a built-in calendar with iCal export so important events can show up in everyones agenda. \index{Trello}
```
Comment (wandernauta) [inline Report/Chapters/Process.tex]: This doesn't really explain what Trello is if you're unfamiliar with it, I think.

File: Report/Chapters/Recommendations.tex [hunk 27]
```diff
@@ -1 +1,18 @@
...
+Sadly, a large number of these ideas were too big or too complicated to be included in this project.
+However, future developers might find these ideas useful as inspiration.
+
+We have compiled a wish list of the features we would like to see in Viskell:
+
+\begin{itemize}
+	\item Circular menu (see \ref{circular_menu} on page \ref{circular_menu}).
+	\item Compact view of functions (for large programs).
+	\item Support for more Haskell functions (e.g. I/O).
+	\item Support for recursion and pattern matching.
```
Comment (wandernauta) [inline Report/Chapters/Recommendations.tex]: Nitpick: what's missing is *explicit* recursion. We have implicit recursion, i.e. through functions like `until`.

File: Report/Chapters/Requirements.tex (hunk 13 above)
Comment (wandernauta) [inline Report/Chapters/Requirements.tex]: I think this is putting it neutrally and without causing offence. In other words, I like it.

File: Report/Chapters/Design.tex (hunk 24 above)
Comment (jjkester) [inline Report/Chapters/Design.tex]: Fixed.

File: Report/Chapters/Requirements.tex [hunk 28]
```diff
@@ -1,26 +1,76 @@
...
+
+\begin{enumerate}
+\item \subsection*{Visual requirements}
+
+\begin{enumerate}
+	\item The structure of the visual program must logically represent the structure of a regular Haskell program.
+		\begin{enumerate}
+			\item Haskell functions are represented by a separate element.
+			\item The program has to be able to show a compact notation for a program using stacked and nested blocks.
+			\item Blocks must be movable to prevent them from overlap.
```
Comment (wandernauta) [inline Report/Chapters/Requirements.tex]: Better: `prevent overlap` or `prevent them from overlapping`.

File: Report/Chapters/Requirements.tex [hunk 29]
```diff
@@ -1,26 +1,76 @@
...
+For this project, the requirements can be divided in three subgroups, namely: visual requirements, functional requirements and multi-touch requirements. In the following subsections each of the groups will be discussed. In some cases a requirement fits in multiple groups; in these cases the best fitting group has been chosen to contain this requirement.
+
+\begin{enumerate}
+\item \subsection*{Visual requirements}
+
+\begin{enumerate}
+	\item The structure of the visual program must logically represent the structure of a regular Haskell program.
+		\begin{enumerate}
+			\item Haskell functions are represented by a separate element.
+			\item The program has to be able to show a compact notation for a program using stacked and nested blocks.
```
Comment (wandernauta) [inline Report/Chapters/Requirements.tex]: We didn't manage this requirement. Do we have to mark that somehow?

File: Report/Chapters/Requirements.tex [hunk 30]
```diff
@@ -1,26 +1,76 @@
...
+			\item Blocks must be movable to prevent them from overlap.
+		\end{enumerate}
+	\item The program must allow functions/blocks to be interconnected.
+		\begin{enumerate}
+			\item Function arguments must be connected individually.
+			\item A block has a single output which can be the final result of the function or the result of a partial application.
+		\end{enumerate}
+	\item The user interface design must be minimalistic.
+	\item The user interface must be resizeable.
```
Comment (wandernauta) [inline Report/Chapters/Requirements.tex]: Or: `work on large (table-sized) screens as well as small (laptops-sized) screens`.

File: Report/Chapters/Requirements.tex [hunk 31]
```diff
@@ -1,26 +1,76 @@
...
+	\item Actions performed by the user must give feedback.
+	\item Errors and warnings (for example type errors) must be easily recognizable.
+	\item In case blocks and lines overlap, lines must be drawn behind the blocks.
```
Comment (wandernauta) [inline Report/Chapters/Requirements.tex]: Do we do this?

File: Report/Chapters/Process.tex (hunk 26 above)
Comment (jjkester) [inline Report/Chapters/Process.tex]: I agree. Do we want to explain Trello to the reader, or let them use Google and check it out themselves?

File: Report/Chapters/Recommendations.tex (hunk 27 above)
Comment (jjkester) [inline Report/Chapters/Recommendations.tex]: You're right.

**Comment by wandernauta** [comment]
Merged! :tada: Thanks a lot.
